pkg/controllers/management/eks/eks_cluster_handler.go

```diff
@@ -190,7 +190,11 @@ func (e *eksOperatorController) onClusterChange(key string, cluster *mgmtv3.Clus
 		// for rancher to validate its ability to do so.
 		addNgMessage := "Cluster must have at least one managed nodegroup."
 		noNodeGroupsOnSpec := len(cluster.Spec.EKSConfig.NodeGroups) == 0
-		noNodeGroupsOnUpstreamSpec := len(cluster.Status.EKSStatus.UpstreamSpec.NodeGroups) == 0
+		noNodeGroupsOnUpstreamSpec := true // default to true in case UpstreamSpec is nil
```

Comment:
By this point the upstream spec should already be set, do we know why that isn't happening? If `noNodeGroupsOnUpstreamSpec` is false then the cluster basically gets set to an error state.